main.py

```diff
@@ -164,30 +237,58 @@ def resize(viewer):
         assert False, "Invalid scene type, %s. Must be one of xx x2222 x442 x642 x333 or xN, " \
                       "where N is a positive integer." % args.type
 
-output_dir = "./output_%s_%s" % (os.path.basename(args.filename), str(int(time.time())))
-output_dir = os.path.realpath(output_dir)
-os.mkdir(output_dir)
 
+print("Generating fds on R2...")
 frustum = scene_parsing.make_frustum(args.filename)
 kt = tiling.KernelTiling(base_kernel, frustum, args.overlap)
+
+if args.ceiling or args.floor:
+    print("Generating fds on R3...")
+
+    vgroup = tiling.FriezeReflectionGroup(args.height, (0, 0, 1), (0, args.height, 0),(0, 0, 0))
+    if args.floor and not args.ceiling:
+        vgroup = tiling.FriezeReflectionGroup(args.height, (0, 0, 1), (0, 0, 0),(0, args.height, 0))
+
+    if args.type == "xx":
+        vbase_kernel = tiling.vLineKernel(args.vradius, (0,0), vgroup, base_kernel, args.ceiling, args.floor)
+
+    elif args.type == "x2222" or args.type == "x442":
+        vbase_kernel = tiling.vSquareKernel(args.vradius, (0,0,0), vgroup, base_kernel, args.ceiling, args.floor)
+
+    elif args.type == "x632" or args.type == "x333":
+        vbase_kernel = tiling.vHexKernel(args.vradius, (0, 0, 0), vgroup, base_kernel, args.ceiling, args.floor)
+
+    kt = tiling.KernelTiling(vbase_kernel, frustum, args.overlap)
+
```

Comment:
This is duplicated logic from lines 202 through 240 of this file. See my comment on `tiling.py` for more info.